design-documents/Payment-Request-API.md

```diff
@@ -0,0 +1,77 @@
+1. A customer chooses one of the available payment methods or fills credit card details.
+1. The browser sends payment details to the chosen payment application (like Google Pay, Apple Pay, etc.).
+1. Payment Application sends data to a payment gateway.
+1. The payment gateway returns a response with token or credit card details (depends on the chosen payment method).
+1. The application sends received data to the browser.
+1. Now, js component can process received payment details before sending to the payment gateway.
+1. A merchant website sends a transaction with payment details to the payment gateway.
+1. Payment Gateway sends the payment transaction to an issuing bank for a verification.
+1. The issuing bank verifies the transaction and sends back a result to the payment gateway.
+1. The payment gateway sends the response to the merchant's website and it completes the order.
+
+## Magento Core and Payment Gateway Modules Responsibilities
+![Responsibilities](./prapi-resp.png)
+
+Magento Open Source implements the general flow of Payment Request API invocation and provides payment method independent
+data such as order items, total and shipping options.
+
+Payment Gateway integration modules responsible for providing valid data for supported payment methods that may be used
+by Payment Request API.
+
+Payment Gateway integration modules provide information about supported payment methods to Magento Open Source framework
+which manages it.
+
+The working prototype located in the public repository for [Payment Improvements](https://github.com/magento/payment-improvements/tree/payment-request-api).
+All technical details related to the implementation and additional design document located in [Payment Improvements Wiki](https://github.com/magento/payment-improvements/wiki) space.
+
+## Checkout Flow with PR API
+Payment Request API doesn't replace standard Magento checkout flow. The main idea is to provide additional, more consistent
+user experience flow which doesn't require any additional payment forms. As not all browsers support PR API, a customer
+still should have a possibility to use standard checkout. In a case, if PR API is supported by a browser, both variants
+should be available. Fortunately, PR API allows to detect is it supported by the browser or not:
+```javascript
+if(window.PaymentRequest) {
+  // Use Payment Request API
+} else {
+  // Fallback to traditional checkout
+  window.location.href = '/checkout/traditional';
+}
+```
+## PCI Compliance and Security
+
+As PR API with basic-card payment method can return credit card data like, CC number, expiration date, CVV, etc. there
+are few important security notes. If merchant's site is compliant with PCI SAQ A, it isn't allowed to handle raw credit
+card data directly. The merchants compliant with PCI DSS or
+[PCI SAQ A-EP](https://www.pcisecuritystandards.org/documents/PCI-DSS-v3_2-SAQ-A_EP.pdf) should not worry about it.
```

Comment:
Most sites, including Magento Cloud, will not be PCI SAQ A-EP compliant.

If the default expectation for the Payment API requires PCI SAQ A-EP, then it will not be usable by the large majority of Commerce installations.

The working assumption needs to be no more than PCI SAQ A compliance, and if the solution does not, it is not viable.

(hope I'm not intruding on things here, but I'm the Architect on the new Magento Payments service being developed, and PCI compliance is something we deal with all the time)

Reply:
@nofuss For now, we are going to use `basic-cards` payment method for development purpose. For production, we are going include only payment methods with tokenization such as Google Pay and Apple Pay. Research how we may use `basic-cards` without the requirement for all Magento instances be PCI compliant is a subject of additional research magento/payment-improvements#44

Reply:
Sounds like a plan, but I would suggest not using the basic-cards payment method for development purposes, if that is the one method that will not be made available in production initially.

That said, I am not very familiar with this project, so please take my comments with a grain of salt.

Reply:
@nofus Decision to use basic-cards for development dictated by a desire not complicate dev env as Apple Pay and Google Pay requires pretty unusual set up for Magento (PHP and JavaScript) engineers

Reply:
You can potentially offload some of the PCI compliance requirements to a 3rd party such as Braintree to make it a more viable solution to a wider audience: https://developers.braintreepayments.com/guides/payment-request/overview

Reply:
@davemacaulay, this suggestion described in the last paragraph of this document. But we are working on the solution agnostic to the specific Payment Gateways, not all merchants use Braintree or Stripe as the payment provider.